sites/default/statement.inc.php

```diff
@@ -102,278 +463,280 @@
 //  [Your billing dept phone]
 
 function create_statement($stmt) {
```

Comment:
On a diff in this function without whitespace, noting the following changes:
``````
@@ -177,7 +537,7 @@ if ($GLOBALS['use_dunning_message']) {
   $label_insinfo = xl('Insurance information on file');
   $label_totaldue = xl('Total amount due');
   $label_payby = xl('If paying by');
- $label_cards = xl('VISA/MC/AMEX/Dis');
+  $label_cards = xl('VISA/MC/Discovery/HSA');
   $label_cardnum = xl('Card');
   $label_expiry = xl('Exp');
   $label_sign = xl('Signature');
@@ -194,12 +554,13 @@ if ($GLOBALS['use_dunning_message']) {
   //  %-25s = left-justified string of 25 characters padded with spaces
   // Note that "\n" is a line feed (new line) character.
   // reformatted to handle i8n by tony
-
-$out  = sprintf("%-30s %-23s %-s\n",$clinic_name,$stmt['patient'],$stmt['today']);
-$out .= sprintf("%-30s %s: %-s\n",$clinic_addr,$label_chartnum,$stmt['pid']);
-$out .= sprintf("%-30s %-s\n",$clinic_csz,$label_insinfo);
-$out .= sprintf("%-30s %s: %-s\n",null,$label_totaldue,$stmt['amount']);
-$out .= "\n\n";
+  $out = "\n\n";
+  $providerNAME = getProviderName($stmt['providerID']);  
+  $out .= sprintf("%-30s %s %-s\n",$clinic_name,$stmt['patient'],$stmt['today']);
+  $out .= sprintf("%-30s %s: %-s\n",$providerNAME,$label_chartnum,$stmt['pid']);
+  $out .= sprintf("%-30s %s\n",$clinic_addr,$label_insinfo);
+  $out .= sprintf("%-30s %-s: %-s\n",$clinic_csz,$label_totaldue,$stmt['amount']);
+  $out .= "\n";
   $out .= sprintf("       %-30s %-s\n",$label_addressee,$label_remitto);
   $out .= sprintf("       %-30s %s\n",$stmt['to'][0],$remit_name);
   $out .= sprintf("       %-30s %s\n",$stmt['to'][1],$remit_addr);
@@ -211,10 +572,10 @@ $out .= sprintf("_______________________________________________________________
   $out .= "\n";
   $out .= sprintf("%-32s\n",$label_payby.' '.$label_cards);
   $out .= "\n";
-$out .= sprintf("%s_____________________  %s______ %s___________________\n",
+  $out .= sprintf("%s_____________________  %s______ %s___________________%s\n\n",
     $label_cardnum,$label_expiry,$label_sign);
+      .$remit_addr.'<br />'
+      .$remit_csz.'
+      </td>
+      </tr></table>';
+
+  $out .= "      </div></div>";
+  $out .= "\014
+  <br /><br />"; // this is a form feed
+  echo $out;
+  $output = ob_get_clean();
+  return $output;
+}
 
 // This function builds a printable statement or collection letter from
 // an associative array having the following keys:
@@ -146,7 +507,6 @@ function create_statement($stmt) {
   // $stmt['level_closed'] <= 3 insurance 4 = patient
 
   if ($GLOBALS['use_dunning_message']) {
-
     if ($stmt['ins_paid'] != 0 || $stmt['level_closed'] == 4)  {
 
       // do collection messages
@@ -177,7 +537,7 @@ if ($GLOBALS['use_dunning_message']) {
   $label_insinfo = xl('Insurance information on file');
   $label_totaldue = xl('Total amount due');
   $label_payby = xl('If paying by');
- $label_cards = xl('VISA/MC/AMEX/Dis');
+  $label_cards = xl('VISA/MC/Discovery/HSA');
   $label_cardnum = xl('Card');
   $label_expiry = xl('Exp');
   $label_sign = xl('Signature');
@@ -194,12 +554,13 @@ if ($GLOBALS['use_dunning_message']) {
   //  %-25s = left-justified string of 25 characters padded with spaces
   // Note that "\n" is a line feed (new line) character.
   // reformatted to handle i8n by tony
-
-$out  = sprintf("%-30s %-23s %-s\n",$clinic_name,$stmt['patient'],$stmt['today']);
-$out .= sprintf("%-30s %s: %-s\n",$clinic_addr,$label_chartnum,$stmt['pid']);
-$out .= sprintf("%-30s %-s\n",$clinic_csz,$label_insinfo);
-$out .= sprintf("%-30s %s: %-s\n",null,$label_totaldue,$stmt['amount']);
-$out .= "\n\n";
+  $out = "\n\n";
+  $providerNAME = getProviderName($stmt['providerID']);  
+  $out .= sprintf("%-30s %s %-s\n",$clinic_name,$stmt['patient'],$stmt['today']);
+  $out .= sprintf("%-30s %s: %-s\n",$providerNAME,$label_chartnum,$stmt['pid']);
+  $out .= sprintf("%-30s %s\n",$clinic_addr,$label_insinfo);
+  $out .= sprintf("%-30s %-s: %-s\n",$clinic_csz,$label_totaldue,$stmt['amount']);
+  $out .= "\n";
   $out .= sprintf("       %-30s %-s\n",$label_addressee,$label_remitto);
   $out .= sprintf("       %-30s %s\n",$stmt['to'][0],$remit_name);
   $out .= sprintf("       %-30s %s\n",$stmt['to'][1],$remit_addr);
@@ -211,10 +572,10 @@ $out .= sprintf("_______________________________________________________________
   $out .= "\n";
   $out .= sprintf("%-32s\n",$label_payby.' '.$label_cards);
   $out .= "\n";
-$out .= sprintf("%s_____________________  %s______ %s___________________\n",
+  $out .= sprintf("%s_____________________  %s______ %s___________________%s\n\n",
     $label_cardnum,$label_expiry,$label_sign);
+      .$remit_addr.'<br />'
+      .$remit_csz.'
+      </td>
+      </tr></table>';
+
+  $out .= "      </div></div>";
+  $out .= "\014
+  <br /><br />"; // this is a form feed
+  echo $out;
+  $output = ob_get_clean();
+  return $output;
+}
 
 // This function builds a printable statement or collection letter from
 // an associative array having the following keys:
@@ -146,7 +507,6 @@ function create_statement($stmt) {
   // $stmt['level_closed'] <= 3 insurance 4 = patient
 
   if ($GLOBALS['use_dunning_message']) {
-
     if ($stmt['ins_paid'] != 0 || $stmt['level_closed'] == 4)  {
 
       // do collection messages
@@ -177,7 +537,7 @@ if ($GLOBALS['use_dunning_message']) {
   $label_insinfo = xl('Insurance information on file');
   $label_totaldue = xl('Total amount due');
   $label_payby = xl('If paying by');
- $label_cards = xl('VISA/MC/AMEX/Dis');
+  $label_cards = xl('VISA/MC/Discovery/HSA');
   $label_cardnum = xl('Card');
   $label_expiry = xl('Exp');
   $label_sign = xl('Signature');
@@ -194,12 +554,13 @@ if ($GLOBALS['use_dunning_message']) {
   //  %-25s = left-justified string of 25 characters padded with spaces
   // Note that "\n" is a line feed (new line) character.
   // reformatted to handle i8n by tony
-
-$out  = sprintf("%-30s %-23s %-s\n",$clinic_name,$stmt['patient'],$stmt['today']);
-$out .= sprintf("%-30s %s: %-s\n",$clinic_addr,$label_chartnum,$stmt['pid']);
-$out .= sprintf("%-30s %-s\n",$clinic_csz,$label_insinfo);
-$out .= sprintf("%-30s %s: %-s\n",null,$label_totaldue,$stmt['amount']);
-$out .= "\n\n";
+  $out = "\n\n";
+  $providerNAME = getProviderName($stmt['providerID']);  
+  $out .= sprintf("%-30s %s %-s\n",$clinic_name,$stmt['patient'],$stmt['today']);
+  $out .= sprintf("%-30s %s: %-s\n",$providerNAME,$label_chartnum,$stmt['pid']);
+  $out .= sprintf("%-30s %s\n",$clinic_addr,$label_insinfo);
+  $out .= sprintf("%-30s %-s: %-s\n",$clinic_csz,$label_totaldue,$stmt['amount']);
+  $out .= "\n";
   $out .= sprintf("       %-30s %-s\n",$label_addressee,$label_remitto);
   $out .= sprintf("       %-30s %s\n",$stmt['to'][0],$remit_name);
   $out .= sprintf("       %-30s %s\n",$stmt['to'][1],$remit_addr);
@@ -211,10 +572,10 @@ $out .= sprintf("_______________________________________________________________
   $out .= "\n";
   $out .= sprintf("%-32s\n",$label_payby.' '.$label_cards);
   $out .= "\n";
-$out .= sprintf("%s_____________________  %s______ %s___________________\n",
+  $out .= sprintf("%s_____________________  %s______ %s___________________%s\n\n",
     $label_cardnum,$label_expiry,$label_sign);
-$out .= sprintf("%-20s %s\n",null,$label_retpay);
   $out .= sprintf("-----------------------------------------------------------------\n");
+  $out .= sprintf("%-20s %s\n",null,$label_retpay);
   $out .= "\n";
   $out .= sprintf("_______________________ %s _______________________\n",$label_pgbrk);
   $out .= "\n";
@@ -223,7 +584,7 @@ $out .= "\n";
 
   // This must be set to the number of lines generated above.
   //
- $count = 23;
+  $count = 25;
   $num_ages = 4;
   $aging = array();
   for ($age_index = 0; $age_index < $num_ages; ++$age_index) {
@@ -247,9 +608,9 @@ else {
     }
 
     $tmp = substr($description, 0, 14);
-  if ($tmp == 'Procedure 9920' || $tmp == 'Procedure 9921')
-   $description = xl('Office Visit');
-
+    if ($tmp == 'Procedure 9920' || $tmp == 'Procedure 9921' || $tmp == 'Procedure 9200' || $tmp == 'Procedure 9201')
+      $description = str_replace("Procedure",xl('Office Visit').":",$description);
+    //92002-14 are Eye Office Visit Codes
 
     $dos = $line['dos'];
     ksort($line['detail']);
@@ -308,11 +669,12 @@ else {
   // Fixed text labels
   $label_ptname = xl('Name');
   $label_today = xl('Date');
- $label_due = xl('Due');
+  $label_due = xl('Amount Due');
   $label_thanks = xl('Thank you for choosing');
- $label_call = xl('Please call if any of the above information is incorrect');
- $label_prompt = xl('We appreciate prompt payment of balances due');
+  $label_call = xl('Please call if any of the above information is incorrect.');
+  $label_prompt = xl('We appreciate prompt payment of balances due.');
   $label_dept = xl('Billing Department');
+  $label_bill_phone = $GLOBALS['billing_phone_number'];
   $label_appointments = xl('Future Appointments').':';
 
   // This is the bottom portion of the page.
@@ -332,7 +694,7 @@ else {
   $out .= sprintf("%-s\n",$label_prompt);
   $out .= "\n";
   $out .= sprintf("%-s\n",$billing_contact);
- $out .= sprintf("  %-s %-25s\n",$label_dept,$billing_phone);
+  $out .= sprintf("  %-s %-25s\n",$label_dept,$label_bill_phone);
   if($GLOBALS['statement_message_to_patient']) {
     $out .= "\n";
     $statement_message = $GLOBALS['statement_msg_text'];
@@ -376,4 +738,5 @@ if($GLOBALS['show_aging_on_custom_statement']) {
 
   return $out;
 }
+
 ?>